sdk/core/azure-core-tracing-opentelemetry/src/samples/AsyncListKeyVaultSecretsSample.md

```diff
@@ -0,0 +1,107 @@
+# Asynchronously list Key Vault secrets with Azure Core Tracing OpenTelemetry
+ 
+Following documentation describes instructions to run a sample program for asynchronously creating and listing secrets of a Key Vault with tracing instrumentation.
+
+## Getting Started
+Sample uses **[opentelemetry-sdk][opentelemetry_sdk]** as implementation package and **[Logging Exporter][logging_exporter]** as exporter.
+### Adding dependencies to your project:
+```xml
+<dependencies>
+    <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-sdk</artifactId>
+        <version>0.2.0-SNAPSHOT</version>
+    </dependency>
+    <dependency>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-security-keyvault-secrets</artifactId>
+        <version>4.0.0</version>
+    </dependency>
+    <dependency>
+        <groupId>com.azure</groupId>
+        <artifactId>azure-core-tracing-opentelemetry</artifactId>
+        <version>1.0.0-beta.1</version>
+    </dependency>
+    <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-exporters-logging</artifactId>
+        <version>0.2.0-SNAPSHOT</version>
+    </dependency>
+</dependencies>
+```
+
+#### Sample demonstrates tracing when asynchronously creating and listing secrets from a Key Vault using [azure-security-keyvault-secrets][azure_keyvault_secrets] client library.
+```java
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.security.keyvault.secrets.SecretAsyncClient;
+import com.azure.security.keyvault.secrets.SecretClientBuilder;
+import com.azure.security.keyvault.secrets.models.KeyVaultSecret;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.sdk.trace.TracerSdkFactory;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+import reactor.util.context.Context;
+
+import java.util.logging.Logger;
+
+import static com.azure.core.util.tracing.Tracer.PARENT_SPAN_KEY;
+import static java.util.logging.Logger.getLogger;
+
+/**
+ * Sample demonstrates tracing how to add and list secrets in a Key vault with tracing enabled with a Logging Exporter.
+ */
+public class Sample {
+
+    private static final Logger LOGGER = getLogger("Sample");
+    private static final Tracer TRACER;
+    private static final TracerSdkFactory TRACER_SDK_FACTORY;
+
+    static {
+        TRACER_SDK_FACTORY = Helper.configureOpenTelemetryAndJaegerExporter(LOGGER);
+        TRACER = TRACER_SDK_FACTORY.get("Azure-OpenTelemetry");
```

Comment:
```suggestion
        TRACER = TRACER_SDK_FACTORY.get("sample");
```

this should be the name of the client application or part of the app (like logging category)

Reply:
I think this would need more thought as now Opentelemetry supports the concept of `Named Tracers` wherein the users can have provider their `Provider` implementation that can return a different `Tracerfactory` instance.
I have this [issue](https://github.com/Azure/azure-sdk-for-java/issues/6247) here to track this .

Reply:
named tracers intention is to describe what is instrumented (like logging category). In this case this is user app that is instrumented which seems to be called "sample". Azure SDK should request named tracer specific to it.

Reply:
So that means we should be taking the `instrumentation name` from the user?

Reply:
nope, we can just give it the same name as sample app :)

You've done this for logger, right? `private static final Logger LOGGER = getLogger("Sample")` can we just do the same for tracer `TRACER_SDK_FACTORY.get("Sample")`?

Reply:
Oh, I thought we should have the same tracer instance between the user and opentelemetry-sdk call from the plugin for it to behave correctly.

As let's say we have a tracer configured with span processors on the user end and in our code  we do `private static final Tracer TRACER = OpenTelemetry.getTracerFactory().get("Azure-OpenTelemetry");` here, it returns us a different instance which does not have any of the user configurations.
Shouldn't both the tracers have same configurations?

Reply:
all configuration is controlled by tracer-factory. tracers it produces are the same except name. Azure SDK should request its own tracer from default factory. 

Users request tracer from the same factory but give it a different name. All processors and exporters are shared. 

So it boils down to shared tracer-factory instance by requesting default one. You MAY request   tracer-factory (or tracer) from user, but I suggest not to do this yet.